cedar-policy-core/src/ast/entity.rs

```diff
@@ -110,6 +110,28 @@ impl EntityUID {
         })
     }
 
+    /// Create an `EntityUID` by parsing a string, which is required to be normalized.
+    /// That is, this constructor will not accept strings with spurious whitespace
+    /// (e.g. `A :: B :: C::"foo"`), Cedar comments (e.g. `A::B::"bar" // comment`), etc. See
+    /// [RFC 9](https://github.com/cedar-policy/rfcs/blob/main/text/0009-disallow-whitespace-in-entityuid.md)
+    /// for more details and justification.
+    ///
+    /// For the version that accepts whitespace, Cedar comments, and etc, use the
+    /// actual `FromStr` implementation for `EntityUID`.
+    pub fn parse_normalized_euid(s: &str) -> Result<Self, Vec<ParseError>> {
```

Comment:
I was wondering we can unify `parse_normalized_euid` and `parse_normalized_name`. For instance, we can define a trait called something like `FromNormalizedStr<T: FromStr>` and give a default implementation. Then, the two types can implement the trait.

Reply:
We could do this. I'll experiment